src/main/java/ch/njol/skript/expressions/ExprRound.java

```diff
@@ -13,44 +9,70 @@
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
 import ch.njol.util.Kleenean;
 import ch.njol.util.Math2;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Rounding")
 @Description("Rounds numbers normally, up (ceiling) or down (floor) respectively.")
-@Examples({"set {var} to rounded health of player",
-		"set line 1 of the block to rounded \"%(1.5 * player's level)%\"",
-		"add rounded down argument to the player's health"})
+@Examples({
+	"set {var} to rounded health of player",
+	"set line 1 of the block to rounded \"%(1.5 * player's level)%\"",
+	"add rounded down argument to the player's health"
+})
 @Since("2.0")
 public class ExprRound extends PropertyExpression<Number, Long> {
+
+	private enum RoundType {
+		FLOOR("(round[ed] down|floored)"),
+		ROUND("round[ed]"),
+		CEIL("(round[ed] up|ceiled)");
+
+		private final String pattern;
+
+		RoundType(String pattern) {
+			this.pattern = pattern;
+		}
+	}
+
+	private static final RoundType[] ROUND_TYPES = RoundType.values();
+
 	static {
-		Skript.registerExpression(ExprRound.class, Long.class, ExpressionType.PROPERTY,
-				"(a|the|) round[ed] down %number%",
-				"(a|the|) round[ed] %number%",
-				"(a|the|) round[ed] up %number%");
+		String[] patterns = new String[ROUND_TYPES.length * 2];
+		for (RoundType roundType : ROUND_TYPES) {
+			patterns[2 * roundType.ordinal()] = "[a|the] " + roundType.pattern + " %numbers%";
+			patterns[(2 * roundType.ordinal()) + 1] = "%numbers% " + roundType.pattern;
+		}
```

Comment:
I would like to see this part be done with Patterns instead, like how CondCompare does it. It maintains the ability to read the patterns at a glance in the code.